clarify collection signature not available from Community Galaxy

samccann opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Fixes #78347
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Docs Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
docs/docsite/rst/user_guide/collections_using.rst
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

samccann committed:
clarify collection signature not available from Community Galaxy

samccann committed:
make it clear

Dule-martins commented on docs/docsite/rst/user_guide/collections_using.rst:
```diff
@@ -273,7 +273,9 @@ Verifying against ``tar.gz`` files is not supported. If your ``requirements.yml`
 ----------------------
 
-If a collection has been signed by the Galaxy server, the server will provide ASCII armored, detached signatures to verify the authenticity of the MANIFEST.json before using it to verify the collection's contents. You must opt into signature verification by :ref:`configuring a keyring <galaxy_gpg_keyring>` for ``ansible-galaxy``, or by providing the path with the ``--keyring`` option.
+If a collection has been signed by the Automation Hub or self-hosted pulp or on-premise Hub server, the server will provide ASCII armored, detached signatures to verify the authenticity of the MANIFEST.json before using it to verify the collection's contents. This option is not available on community Galaxy.
```
I'm assuming "Automation Hub, self-hosted pulp, and on-premise hub server" are all different servers that provide ASCII armored?

Still don't understand the difference between the Galaxy server and those mentioned above.

s-hertel replied:
`Galaxy` refers to galaxy.ansible.com.

That's not what was meant here, hence the fix, but the docs use `Galaxy server` often even when it may mean/include AH/pulp, perhaps since we have configuration options called `GALAXY_SERVER` and `GALAXY_SERVER_LIST` (which are used to configure Ansible with any galaxy server, not just Galaxy).

Automation hub is https://cloud.redhat.com/api/automation-hub/.

samccann replied:
Yes thanks to both of you. This requires a bit more thought. We have :

- Galaxy - which historically meant galaxy.ansible.com
- Automation Hub,  -which is a Red Hat product
- selfhosted (which can be Automation Hub, and I think what the repository calls galaxyNG )

I don't actually know what pulp is, outside of galaxyNG self-hosted?  But I think what I need to do is start a conversation on naming, because we will definitely have an overalp of galaxy.ansible.com and galaxyNG to deal with. I'll put this PR in WIP for now.

samccann replied:
Looks like we've resolve this to:

- Galaxy - refers to galaxy.ansible.com
- Self-hosted Galaxy refers to the GalaxyNG codebase

I'll update the PR and add this to the glossary as well.

samccann committed:
update to use self-hosted Galaxy to refer to GalaxyNG

samccann committed:
add glossary entry for galaxyNG

samccann committed:
restructure and add small developer section

samccann committed:
clean up nits

samccann committed:
more cleanup

samccann commented:
@s-hertel @Dule-mart - okay in true writer fashion.... I got carried away with edits :-)  Can you take another look and provide feedback etc?

What I did was:
 - moved the install section out of that snippet file.  the snippet files were my too-clever-for-my-own-good way to reuse this content in the 'using collections' section and in the galaxy section. Signing isn't part of community Galaxy so had to move it out. 
 - Combined some of the content from the 'signature verification' section into the new installing signed collections section as it seemed a bit of a repeat. I left the verification section as only verification (aka post install). 
 - added a glossary term for galaxyNG and also stuck a tiny mention about creating signed collections in the collection developer section.

I feel like the 'detached signatures' stuff is still kind of floating around and might need some more detail? Let me know what y'all think.

s-hertel commented on docs/docsite/rst/dev_guide/developing_collections_distributing.rst:
```diff
@@ -179,6 +180,22 @@ For more information on the :file:`galaxy.yml` file, see :ref:`collections_galax
+
+You can include a GnuPG signature with your collection on a self-hosted Galaxy server. See `Enabling collection signing <https://galaxyng.netlify.app/config/collection_signing/>`_ for details.
+
+You can also manually generate detached signatures for a collection using the ``gpg`` CLI using the following step. This step assume you have generated a GPG private key, but do not cover this process.
```
This example is the same as https://galaxyng.netlify.app/config/collection_signing/#manually-signing-a-collection, just in one line but the `also` in this sentence makes it sound like a third option.

s-hertel commented on docs/docsite/rst/user_guide/collections_using.rst:
```diff
@@ -270,54 +345,39 @@ In addition to the ``namespace.collection_name:version`` format, you can provide
    ansible-galaxy collection verify -r requirements.yml --keyring ~/.ansible/pubring.kbx
 
-When a collection is installed from a Galaxy server, the signatures provided by the server to verify the collection's authenticity are saved alongside the installed collections. This data is used to verify the internal consistency of the collection without querying the Galaxy server again when the ``--offline`` option is provided.
+When a collection is installed from a self-hosted Galaxy server, the signatures provided by the server to verify the collection's authenticity are saved alongside the installed collections. This data is used to verify the internal consistency of the collection without querying the self-hosted Galaxy server again when the ``--offline`` option is provided.
```
self-hosted doesn't cover Automation Hub. I liked the original changes you made to this section.

samccann replied:
I'm less inclined to keep talking about AH here in the community docs as they have their own customer docs now. I can check with that team to see if they have this covered over on access.redhat.com yet tho

s-hertel replied:
Isn't it general ansible-galaxy documentation though, which is not community-specific?

s-hertel replied:
What do you think about `Pulp 3 (v3) Galaxy server` instead of `self-hosted Galaxy server`? That would keep it general. We could have a link to https://docs.pulpproject.org/pulp_ansible/ from the glossary. We would need to specify "excluding community Galaxy" for the signature stuff.

samccann replied:
Hmm good point on this being ansible-galaxy documentation that is likely not on access.redhat.com. I'll add AH back.  I'm I'm not sure if the pulp docsite is better/clearer than pointing to the GalaxyNG docsite that this PR does now.

So the original commit had `Automation Hub or self-hosted pulp or on-premise Hub server`. That to me misses out on galaxyNG (oh how I wish I could just call it galaxy NG :-).

So I'll ask the AH writing team if I can just call it Automation Hub, and then we can decide what to call the galaxyNG version.  Can you explain why Pulp 3 (v3) Galaxy is better than self-hosted Galaxy?

s-hertel replied:
Pulp 3 (v3) covers AH and Galaxy NG. Perhaps it is not helpful in adding clarity to user docs, but it's more accurate. `ansible-galaxy` doesn't care whether a server is AH/NG/neither, it just needs to follow the v3 spec https://docs.pulpproject.org/pulp_ansible/restapi.html to have ansible-galaxy signature functionality.

samccann replied:
Ends up the AH on-prem doesn't fully support this feature, though when it does, the official name is Private Automation Hub.  That said, this is being documented separately for customers so I'll focus this as much as possible on Galaxy NG, which we can call Pulp 3 Galaxy.

samccann replied:
ok as I edit, I realize we already created a 'common term' for all of these besties  - a distribution server. So I'll go with  that and update the docs to match (as well as the table that lists the current options). More to come.

s-hertel commented on docs/docsite/rst/user_guide/collections_using.rst:
```diff
@@ -30,6 +30,81 @@ Installing collections with ``ansible-galaxy``
+Installing collections with signature verification
+---------------------------------------------------
+
+If a collection has been signed by a :term:`self-hosted Galaxy` server, the server will provide ASCII armored, detached signatures to verify the authenticity of the ``MANIFEST.json`` before using it to verify the collection's contents. This option is not available on community Galaxy.
```
This section has the same issue with `self-hosted`, but I like the flow of it.

samccann committed:
call it distribution server

samccann commented on docs/docsite/rst/reference_appendices/glossary.rst:
```diff
@@ -33,7 +33,7 @@ when a term comes up on the mailing list.
 
     Ansible Galaxy
-        An `online resource <galaxy.ansible.com>`_ for finding and sharing Ansible community content. Also, the command-line utility that lets users install individual Ansible Collections, for example`` ansible-galaxy install community.crypto``.
+        An `online distribution server <galaxy.ansible.com>`_ for finding and sharing Ansible community content, somtimes referred to as community Galaxy. Also, the command-line utility that lets users install individual Ansible Collections, for example ``ansible-galaxy collection install community.crypto``.
```
```suggestion
        An `online distribution server <galaxy.ansible.com>`_ for finding and sharing Ansible community content, sometimes referred to as community Galaxy. Also, the command-line utility that lets users install individual Ansible Collections, for example ``ansible-galaxy collection install community.crypto``.
```

samccann commented on docs/docsite/rst/user_guide/collections_using.rst:
```diff
@@ -7,7 +7,7 @@ Using collections
 Collections are a distribution format for Ansible content that can include playbooks, roles, modules, and plugins. As modules move from the core Ansible repository into collections, the module documentation will move to the :ref:`collections pages <list_of_collections>`.
 
-You can install and use collections through `Ansible Galaxy <https://galaxy.ansible.com>`_.
+You can install and use collections through `Ansible Galaxy <https://galaxy.ansible.com>`_ or a `self-hosted Galaxy server <https://galaxyng.netlify.app/>`_.
```
```suggestion
You can install and use collections through a distribution server, such as `Ansible Galaxy <https://galaxy.ansible.com>`_ or a `Pulp 3 Galaxy server <https://galaxyng.netlify.app/>`_.
```

samccann commented on docs/docsite/rst/user_guide/collections_using.rst:
```diff
@@ -270,54 +345,39 @@ In addition to the ``namespace.collection_name:version`` format, you can provide
+-----------------------------
 
-.. code-block:: bash
+If a collection has been signed by a :term:`distribution server`, the server will provide ASCII armored, detached signatures to verify the authenticity of the MANIFEST.json before using it to verify the collection's contents. TThis option is not available on all distribution servers. See :ref:`distributing_collections` for a table listing which servers support collection signing. See :ref:`instal ...
```
```suggestion
If a collection has been signed by a :term:`distribution server`, the server will provide ASCII armored, detached signatures to verify the authenticity of the MANIFEST.json before using it to verify the collection's contents. This option is not available on all distribution servers. See :ref:`distributing_collections` for a table listing which servers support collection signing. See :ref:`installing_signed_collections` for how to verify a signed collection when you install it.
```

samccann committed:
Apply suggestions from code review

samccann commented:
@s-hertel @Dule-mart - I've updated to use 'distribution server' and a table that lists the current options to help make things clearer, I hope. Can you take another look please?

samccann commented:
/azp run
